### PR TITLE
chore(check_rfcs): update to give warning on failure instead of throwing error

### LINKED ISSUE #5917: Installation fails due to GitHub rate limit issues
### Current behaviour

Occasionally, when we build a project using `carbon-react` it fails with the following error on installation:

```
RequestError [HttpError]: API rate limit exceeded for 51.103.107.49. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
#18 78.84     at /xtrem/app/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:86:21
#18 78.84     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
#18 78.84     at async getOpenRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:20:20)
#18 78.84     at async checkRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:42:20) {
#18 78.84   status: 403,
#18 78.84   response: {
#18 78.84     url: '[https://api.github.com/repos/sage/carbon/pulls'](https://api.github.com/repos/sage/carbon/pulls%27),
#18 78.84     status: 403,
#18 78.84     headers: {
```

This is very inconvenient if it happens in our nightly builds (it happened twice in the last week), because then our environments are not updated next day and our nightly integration tests are not executed.

### Expected behaviour

The installation should not stop even if this call fails, given this call seems to be for information purposes only.

### CodeSandbox or Storybook URL

N/A

### JIRA Ticket (Sage Only)

_No response_

### Suggested Solution

Catch any HTTP error that happens during the NPM postinstall scripts so they don't block the installation.
OR
Skip the call when a specific env var is defined. 

### Carbon Version

114.17.1

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Other

### What Operating System are you seeing the problem on?

Other

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by Parsium**
closes #5917

### Proposed behaviour

Update `check_rfcs` script so:
- Content is more recognisable in the terminal
- Outputs a warning in the terminal upon failure to retrieve data on open RFCs. This will prevent consumers from receiving an http error which exits the installation script. 

Screenshot 2023-03-22 at 15 28 56
Screenshot 2023-03-22 at 15 29 03

### Current behaviour

If `check_rfcs` script fails to retrieve RFCs, any raised http error will not be caught and handled properly - causing the installation script to exit. A commonly encountered http error is when the GitHub API rate limit is exceeded:

```shell
RequestError [HttpError]: API rate limit exceeded for 51.103.107.49. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
#18 78.84     at /xtrem/app/node_modules/@octokit/core/node_modules/@octokit/request/dist-node/index.js:86:21
#18 78.84     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
#18 78.84     at async getOpenRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:20:20)
#18 78.84     at async checkRfcs (/xtrem/app/node_modules/carbon-react/scripts/check_rfcs/check_rfcs.js:42:20) {
#18 78.84   status: 403,
#18 78.84   response: {
#18 78.84     url: '[https://api.github.com/repos/sage/carbon/pulls'](https://api.github.com/repos/sage/carbon/pulls%27),
#18 78.84     status: 403,
#18 78.84     headers: {
```

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

Run `npm run postinstall` to check that the script is called correctly without raising an error.

### CODE REVIEW

File: scripts/check_rfcs/check_rfcs.spec.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 const ci = require("ci-info");
 const nock = require("nock");
-const checkRfcs = require("./check_rfcs");
+const checkRfcs = require("./check_rfcs.js");
```
Comment (edleeks87) [inline scripts/check_rfcs/check_rfcs.spec.js:3]:
```suggestion
const checkRfcs = require("./check_rfcs");
```

File: scripts/check_rfcs/check_rfcs.js [hunk 2]
```diff
@@ -32,26 +31,42 @@ const getOpenRfcs = async () => {
...
     return;
   }
 
-  const openRfcs = await getOpenRfcs();
+  try {
+    const openRfcs = await getOpenRfcs();
 
-  if (openRfcs.length > 0) {
-    console.log("\ncarbon-react currently has open RFCs:");
+    if (openRfcs.length < 1) return;
```
Comment (edleeks87) [inline scripts/check_rfcs/check_rfcs.js:44]:
```suggestion
    if (!openRfcs.length) return;
```

File: scripts/check_rfcs/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const checkRfcs = require("./check_rfcs.js");
```
Comment (edleeks87) [inline scripts/check_rfcs/index.js:1]:
```suggestion
const checkRfcs = require("./check_rfcs");
```

File: scripts/check_rfcs/check_rfcs.spec.js [hunk 4]
```diff
@@ -57,39 +50,46 @@ describe("checkRfcs script", () => {
...
+    process.env.NODE_ENV = "NOT-TEST";
+    await checkRfcs();
+
+    expect(consoleLogMock).not.toHaveBeenCalled();
+    process.env.NODE_ENV = OLD_ENV;
+  });
 
-      expect(consoleLogMock).toHaveBeenCalled();
+  describe("when not being run in CI", () => {
+    it("if script retrieves pull request data, should run and call console log once", async () => {
```
Comment (nineteen88) [inline scripts/check_rfcs/check_rfcs.spec.js:65]:
```suggestion
    it("should run and call console log once when script retrieves pull request data, ", async () => {
```
Super minor and not expecting you to change this as we do this quite a lot in the codebase but the `it` should make contextual sense in my opinion.

**Review by nineteen88 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**